ladybug/datacollection.py

```diff
@@ -392,7 +369,7 @@ def interpolate_data(self, timestep, cumulative=False):
                 _data[i].value = d.value / timestep
 
         # shift data if half-hour interpolation has been selected.
-        if self.middle_hour is True:
+        if self.header.middle_hour is True:
```

Comment:
You need a check here to ensure header is not None before checking for middle_hour unless we are generating an empty header if not provided.